<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>670</width>
    <height>354</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    background-color: #e6f0f7; 
    color: #333333;
    font-family: &quot;Segoe UI&quot;, Arial;
    font-size: 12px;
}

QTabWidget::pane {
    border: 1px solid #c2d6eb;
    margin: 0px;
}

QTabBar::tab {
    background: #d0e0f0;
    color: #555555;
    padding: 8px 15px;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
}

QTabBar::tab:selected {
    background: #e6f0f7;
    color: #003366;
    border-bottom: 2px solid #6A5ACD;
}

QTableView {
    background: #ffffff; 
    gridline-color: #e0e0e0; 
    border: 1px solid #e0e0e0;
    selection-background-color: #6A5ACD;
    selection-color: white; 
}

QHeaderView::section {
    background-color: #ffffff;
    color: #1c2fff; 
    padding: 4px;
    border: 1px solid #e0e0e0;
    font-weight: bold;
}

QPushButton {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #6A5ACD, stop:1 #4169E1
    );
    color: white;
    border: none;
    padding: 8px 15px;
    border-radius: 6px;
    min-width: 80px;
}

QPushButton:hover {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #7B68EE, stop:1 #4876FF
    );
}

QPushButton:pressed {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #483D8B, stop:1 #0000CD
    );
}

QLineEdit {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #E0FFFF, stop:1 #D0E0F0
    );
    border: 1px solid #A0C4E3;
    padding: 5px;
    border-radius: 6px;
    selection-background-color: #6A5ACD;
    color: #1c2fff; 
}

QLineEdit:focus {
    border: 1px solid #6A5ACD;
    background: #F0F8FF;
}

QLabel {
    color: #1c2fff; 
    font-weight: bold;
    padding: 2px 5px;
    background-color: transparent;
}

QLabel::disabled {
    color: #808080;
}

QMenu {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #E0FFFF, stop:1 #C0E0F7
    );
    color: #000000;
    border: 1px solid #6A5ACD;
    margin-top: 2px;
    border-radius: 4px;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

QMenu::item:selected {
    background-color: #6A5ACD;
    color: white;
}

QToolBar {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #E0FFFF, stop:1 #C0E0F7
    );
    color: #1c2fff;
    border: none;
    spacing: 6px;
    padding: 4px;
    border-radius: 6px;
}

QMenuBar {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #D0E6F8, stop:1 #A0C4E3
    );
    color: #1c2fff;
    border: none;
    padding: 4px;
    border-radius: 6px;
}

QMenuBar::item:selected {
    background-color: #B0D4EC;
    color: white;
}

QScrollBar:vertical {
    background: #f0f7ff;
    width: 10px;
}

QScrollBar::handle:vertical {
    background: qlineargradient(
        x1:0, y1:0, x2:0, y2:1,
        stop:0 #6A5ACD, stop:1 #4169E1
    );
    min-height: 20px;
    border-radius: 5px;
}

QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
    background: #d0e0f0;
    border: 1px solid #c2d6eb;
    border-radius: 3px;
}

QScrollBar::add-line:vertical:hover, QScrollBar::sub-line:vertical:hover {
    background: #B0D4EC;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <item>
     <layout class="QHBoxLayout" name="bottomLayout"/>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="8" column="4">
         <widget class="QPushButton" name="calculate">
          <property name="text">
           <string>подсчёт</string>
          </property>
         </widget>
        </item>
        <item row="6" column="4">
         <widget class="QPushButton" name="clear_table">
          <property name="text">
           <string>очистка таблицы</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QPushButton" name="add_characteristic">
          <property name="text">
           <string>добавление характеричтики</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <widget class="QPushButton" name="delete_product">
          <property name="text">
           <string>удалениие продукта</string>
          </property>
         </widget>
        </item>
        <item row="8" column="2">
         <widget class="QLineEdit" name="lineEdit"/>
        </item>
        <item row="6" column="2">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Путь: нет</string>
          </property>
         </widget>
        </item>
        <item row="8" column="3">
         <widget class="QPushButton" name="delete_characteristic">
          <property name="text">
           <string>удаление характеристики</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QPushButton" name="add_product">
          <property name="styleSheet">
           <string notr="true">QPushButton {&quot;
background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
stop:0 #6A5ACD, stop:1 #4169E1);
color: white;&quot;&quot;   border: none;
padding: 8px 15px;
border-radius: 4px;
font-weight: bold;
}
QPushButton:hover {
background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
stop:0 #7B68EE, stop:1 #4876FF);
}
QPushButton:pressed {
background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
stop:0 #483D8B, stop:1 #0000CD)
}</string>
          </property>
          <property name="text">
           <string>добавление продукта</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTableView" name="tableView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>670</width>
     <height>30</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QMenu" name="menu">
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Button">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="ButtonText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="PlaceholderText">
        <brush brushstyle="SolidPattern">
         <color alpha="128">
          <red>28</red>
          <green>47</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Button">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="ButtonText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="PlaceholderText">
        <brush brushstyle="SolidPattern">
         <color alpha="128">
          <red>28</red>
          <green>47</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>28</red>
          <green>47</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Button">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>28</red>
          <green>47</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="ButtonText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>28</red>
          <green>47</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="LinearGradientPattern">
         <gradient startx="0.000000000000000" starty="0.000000000000000" endx="0.000000000000000" endy="1.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
          <gradientstop position="0.000000000000000">
           <color alpha="255">
            <red>224</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </gradientstop>
          <gradientstop position="1.000000000000000">
           <color alpha="255">
            <red>192</red>
            <green>224</green>
            <blue>247</blue>
           </color>
          </gradientstop>
         </gradient>
        </brush>
       </colorrole>
       <colorrole role="PlaceholderText">
        <brush brushstyle="SolidPattern">
         <color alpha="128">
          <red>28</red>
          <green>47</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="title">
     <string>файл</string>
    </property>
    <addaction name="saveDB"/>
    <addaction name="readDB"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="about_me"/>
  </widget>
  <action name="saveDB">
   <property name="text">
    <string>сохранить</string>
   </property>
   <property name="toolTip">
    <string>сохранить файл</string>
   </property>
   <property name="font">
    <font>
     <bold>false</bold>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="readDB">
   <property name="text">
    <string>читать</string>
   </property>
   <property name="toolTip">
    <string>читать файл</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="about_me">
   <property name="text">
    <string>о авторе</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
